example_cpp_smart_card_client_app/src/chrome_certificate_provider/bridge-integrationtest.js

```diff
@@ -0,0 +1,104 @@
+/** @license
+ * Copyright 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+goog.require('SmartCardClientApp.CertificateProviderBridge.Backend');
+goog.require('GoogleSmartCard.IntegrationTestController');
+goog.require('GoogleSmartCard.Logging');
+goog.require('goog.Promise');
+goog.require('goog.promise.Resolver');
+goog.require('goog.testing');
+goog.require('goog.testing.jsunit');
+
+goog.setTestOnly();
+
+goog.scope(function() {
+
+const GSC = GoogleSmartCard;
+const CertificateProviderBridge = SmartCardClientApp.CertificateProviderBridge;
+
+/** @type {GSC.IntegrationTestController?} */
+let testController;
+/** @type {CertificateProviderBridge.Backend?} */
+let bridgeBackend;
+/** @type {!goog.promise.Resolver<!chrome.certificateProvider.SetCertificatesDetails>} */
+let setCertificatesApiExpectation;
+
+/**
+ * Sets up stub implementation of the chrome.certificateProvider API.
+ */
+function setUpApiStubs() {
+  testController.propertyReplacer.set(chrome, 'certificateProvider', {
+    'setCertificates': stubSetCertificatesApi,
+  });
+}
+
+/**
+ * Sets up stub implementation of the legacy chrome.certificateProvider API (the
+ * one that existed before the setCertificates() method was introduced).
+ */
+function setUpLegacyApiStubs() {
+  testController.propertyReplacer.set(chrome, 'certificateProvider', {});
+}
+
+/**
+ * Stub implementation of the chrome.certificateProvider.setCertificates() API.
+ * @param {!chrome.certificateProvider.SetCertificatesDetails} details
+ * @param {function()=} callback
+ */
+function stubSetCertificatesApi(details, callback) {
+  setCertificatesApiExpectation.resolve(details);
+  if (callback)
+    callback();
+}
+
+goog.exportSymbol('testChromeCertificateProviderApiBridge', {
+  setUp: function() {
+    testController = new GSC.IntegrationTestController();
+    setCertificatesApiExpectation = goog.Promise.withResolver();
```

Comment:
nit: I think this should be initialized either in setUpApiStubs or in testSetCertificates_empty